### ISSUE TITLE
[Bug]: Duplicate "provides" packages are not uninstalled

### ISSUE THREAD

**Issue by Samisafool** [CyberShadow/aconfmgr#144]
### General description of the problem:

So for example say you have `xdg-desktop-portal{,-kde}` installed and in config. If you also install `xdg-desktop-portal-wlr` (and mark it as dependency), it will not get uninstalled on the next `aconfmgr apply`.

### Steps to reproduce the problem:

1. Use the provided config and apply it
2. Install xdg-desktop-portal-wlr (or any other flavour)
3. Run `aconfmgr apply`.

### Configuration:

```bash
AddPackage xdg-desktop-portal
AddPackage xdg-desktop-portal-kde
```

### Expected result:

It should remove xdg-desktop-portal-wlr because that is not in the config.

### Actual result:

It does not remove it, instead merely unpinning it if it was marked as explicit.

### Log:

```text
None
```

### Additional context:

So basically, `xdg-desktop-portal-{gnome,gtk,kde,wlr}` are backends for `xdg-desktop-portal`. They can be installed side-by-side, and they all are indirect dependencies of `xdg-desktop-portal` since they all provide `xdg-desktop-portal-impl`. However, they do not conflict. So aconfmgr ignores all of them.

**Comment by hvitoi** [comment]
I faced this same issue with `jdk{8,11,17}-openjdk`
